tests/WE2E/run_WE2E_tests.py

```diff
@@ -0,0 +1,447 @@
+        logging.info("calling function that monitors jobs, prints summary")
+        monitor_file = monitor_jobs(monitor_yaml, args.debug)
+
+        logging.info("All experiments are complete")
+        logging.info(f"Summary of results available in {monitor_file}")
+
+
+
+
+
+def check_tests(tests: list) -> list:
+    """
+    Function for checking that all tests in a provided list of tests are valid
+
+    Args:
+        tests        : List of potentially valid test names
+    Returns:
+        tests_to_run : List of config files corresponding to test names
+    """
+
+    testfiles = glob.glob('test_configs/**/config*.yaml', recursive=True)
+    tests_to_run=[]
+    for test in tests:
+        # Skip blank/empty testnames; this avoids failure if newlines or spaces are included
+        if test.isspace():
+            continue
+        match=False
+        # Search for exact config file name to avoid accidental partial matches
+        test_config='config.' + test.rstrip() + '.yaml'
+        for testfile in testfiles:
+            if test_config in testfile:
+                logging.debug(f"found test {test}")
+                match=True
+                tests_to_run.append(os.path.abspath(testfile))
+        if not match:
+            raise Exception(f"Could not find test {test}")
+    # Because some test files are symlinks to other tests, check that we don't
+    # include the same test twice
+    for testfile in tests_to_run.copy():
+        if os.path.islink(testfile):
+            if os.path.realpath(testfile) in tests_to_run:
+                logging.warning(dedent(f"""WARNING: test file {testfile} is a symbolic link to a
+                                test file ({os.path.realpath(testfile)}) that is also included in the
+                                test list. Only the latter test will be run."""))
+                tests_to_run.remove(testfile)
+    if len(tests_to_run) != len(set(tests_to_run)):
+        logging.warning("\nWARNING: Duplicate test names were found in list. Removing duplicates and continuing.\n")
+        tests_to_run = list(set(tests_to_run))
+    return tests_to_run
+
+
+
+def check_task_get_extrn_ics(cfg: dict, mach: dict, dflt: dict) -> dict:
+    """
+    Function for checking and updating various settings in task_get_extrn_ics section of test config yaml
+
+    Args:
+        cfg  : Dictionary loaded from test config file
+        mach : Dictionary loaded from machine settings file
+        dflt : Dictionary loaded from default config file
+    Returns:
+        cfg_ics : Updated dictionary for task_get_extrn_ics section of test config
+    """
+
+    #Make our lives easier by shortening some dictionary calls
+    cfg_ics = cfg['task_get_extrn_ics']
+
+    # If RUN_TASK_GET_EXTRN_ICS is false, do nothing and return
+    if 'workflow_switches' in cfg:
+        if 'RUN_TASK_GET_EXTRN_ICS' in cfg['workflow_switches']:
+            if cfg['workflow_switches']['RUN_TASK_GET_EXTRN_ICS'] is False:
```

Comment:
The example you give will fail if the key `workflow_switches` is not set, and it will return if RUN_TASK_GET_EXTRN_ICS is not set, neither of which is acceptable. Unfortunately because we can't guarantee that this key exists, *and* we need to check that *if* the key cfg['workflow_switches']['RUN_TASK_GET_EXTRN_LBCS'] is set it is *false* (if it is not set, it defaults to `True`), this logic necessarily needs to be a little convoluted.

Reply:
Oh. Sure. What about:

```
if cfg.get('workflow_switches', {}).get('RUN_TASK_GET_EXTRN_ICS', True) is False:
    return cfg_ics
```

The logic should return if `workflow_switches` is set and `RUN_TASK_GET_EXTRN_ICS` is set to something that evaluates to False.

If `workflow_switches` or `RUN_TASK_GET_EXTRN_ICS` isn't set, or `RUN_TASK_GET_EXTRN_ICS` evaluates to something not False, it does not return.